src/main/thrift/parquet.thrift

```diff
@@ -162,19 +162,48 @@ enum ConvertedType {
   BSON = 20;
 
   /**
+   * @Deprecated: use INTERVAL_YEAR_MONTH or INTERVAL_DAY_TIME
+   * since the SQL standard defines either YEAR_MONTH or DAY_TIME unit.
+   * This is deprecated in favor of those 2 types
+   *
    * An interval of time
-   * 
+   *
    * This type annotates data stored as a FIXED_LEN_BYTE_ARRAY of length 12
    * This data is composed of three separate little endian unsigned
    * integers.  Each stores a component of a duration of time.  The first
    * integer identifies the number of months associated with the duration,
    * the second identifies the number of days associated with the duration
-   * and the third identifies the number of milliseconds associated with 
+   * and the third identifies the number of milliseconds associated with
    * the provided duration.  This duration of time is independent of any
    * particular timezone or date.
    */
   INTERVAL = 21;
-  
+
+  /**
+   * An interval of time with a year-month unit.
+   *
+   * This type annotates data stored as an INT32
+   * This data is stored as a little endian unsigned
+   * integer identifying the number of months associated with the duration.
+   * This duration of time is independent of any
+   * particular timezone or date.
+   */
+  INTERVAL_YEAR_MONTH = 22;
+
+  /**
+   * An interval of time with days-milliseconds unit
+   *
+   * This type annotates data stored as a FIXED_LEN_BYTE_ARRAY of length 8
+   * This data is composed of two separate little endian unsigned
+   * integers.  Each stores a component of a duration of time.
+   *
+   * The first identifies the number of days associated with the duration
+   * and the second identifies the number of milliseconds associated with
+   * the provided duration.  This duration of time is independent of any
+   * particular timezone or date.
+   */
+  INTERVAL_DAY_TIME = 23;
```

Comment:
@rdblue, there are basically two approaches:

1) Treat days and seconds as two dimensions. In this case seconds cannot be constrained by 86,400 (otherwise you cannot express 25 hour days on a DST change). Addition and multiplication by a scalar works per dimension. 0 days, 22 hours + 0 days, 4 hours = 0 days, 26 hours.
2) Treat one day as 24 hours making this one dimension: a scalar of seconds (which I consider the SQL way, but it seems you are not convinced that this is the SQL standard semantic). 0 days 22 hours + 0 days 4 hour = 1 day 2 hours. In this case the storage can be `days + 86400000 + ms` to make the ordering and arithmetic easier. While it could be still stored as 2 values, it would complicate the implementation and hamper the encoding. Which was my original point.

Going forward with option 1) is fine (as long as it is clearly documented), but then I don't see the advantage over the existing interval type with this is deprecating.